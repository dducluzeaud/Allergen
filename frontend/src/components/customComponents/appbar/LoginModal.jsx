import React, { useState } from 'react';
import { Formik, Form, Field } from 'formik';
import PropTypes from 'prop-types';
import * as Yup from 'yup';

import Typography from '@material-ui/core/Typography';
import Grid from '@material-ui/core/Grid';
import Button from '@material-ui/core/Button';
import { makeStyles } from '@material-ui/core/styles';

import { login } from 'utils/api/User';

import CustomInput from 'components/customComponents/form/CustomInput';
import CenteredModal from '../CenteredModal';

const useStyles = makeStyles({
  field: {
    marginBottom: '1em',
  },
  button: {
    marginTop: '1em',
    marginBottom: '1em',
  },
});

const LoginSchema = Yup.object().shape({
  username: Yup.string().required("Nom d'utilisateur requis"),
  password: Yup.string().required('Mot de passe requis'),
});

const LoginModal = (props) => {
  const classes = useStyles();
  const [error, setError] = useState();

  const handleLogin = async (data, { setSubmitting }) => {
    try {
      const { onClose } = props;
      await login(data);
      onClose();
    } catch (e) {
      console.log(e, 'YOLO');
      const { response } = e;
      if (response.status === 400) setError(response.data.non_field_errors[0]);
      if (response.status === 401) setError('Aucun compte trouv√© avec ces informations');
    } finally {
      setSubmitting(false);
    }
  };

  return (
    <CenteredModal {...props}>
      <Typography variant="h6" color="inherit">
        Connecter vous!
      </Typography>
      <Formik
        initialValues={{
          username: '',
          password: '',
        }}
        validationSchema={LoginSchema}
        onSubmit={(values, actions) => handleLogin(values, actions)}
      >
        {({ isSubmitting }) => (
          <Form>
            <Grid container direction="column" justify="center" alignItems="center">
              <Field
                name="username"
                component={CustomInput}
                className={classes.field}
                placeholder="Nom d'utilisateur"
              />
              <Field
                name="password"
                type="password"
                component={CustomInput}
                className={classes.field}
                placeholder="Mot de passe"
              />
              <Button
                variant="contained"
                size="large"
                color="primary"
                type="submit"
                className={classes.button}
                disabled={isSubmitting}
              >
                Me connecter
              </Button>
              {error !== '' && (
                <Typography align="center" variant="body1" color="error">
                  {error}
                </Typography>
              )}
            </Grid>
          </Form>
        )}
      </Formik>
    </CenteredModal>
  );
};

LoginModal.propTypes = {
  onClose: PropTypes.func.isRequired,
};

export default LoginModal;
