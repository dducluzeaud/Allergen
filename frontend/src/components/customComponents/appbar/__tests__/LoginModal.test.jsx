import React from 'react';
import { shallow } from 'enzyme';
import LoginModal from '../LoginModal';

describe('<LoginModal />', () => {
  it('Match snapshot', () => {
    const loginComp = shallow(<LoginModal />);
    expect(loginComp).toMatchSnapshot();
  });
});
