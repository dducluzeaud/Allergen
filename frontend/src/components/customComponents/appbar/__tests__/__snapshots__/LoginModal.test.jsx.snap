// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginModal /> Match snapshot 1`] = `
<CenteredModal
  onClose={null}
  visible={null}
>
  <WithStyles(ForwardRef(Typography))
    color="inherit"
    variant="h6"
  >
    Connecter vous!
  </WithStyles(ForwardRef(Typography))>
  <Formik
    enableReinitialize={false}
    initialValues={
      Object {
        "password": "",
        "username": "",
      }
    }
    isInitialValid={false}
    onSubmit={[Function]}
    validateOnBlur={true}
    validateOnChange={true}
    validationSchema={
      ObjectSchema {
        "_blacklist": RefSet {
          "list": Set {},
          "refs": Map {},
        },
        "_conditions": Array [],
        "_defaultDefault": [Function],
        "_deps": Array [],
        "_excludedEdges": Array [],
        "_exclusive": Object {},
        "_mutate": undefined,
        "_nodes": Array [
          "password",
          "username",
        ],
        "_options": Object {
          "abortEarly": true,
          "recursive": true,
        },
        "_type": "object",
        "_typeError": [Function],
        "_whitelist": RefSet {
          "list": Set {},
          "refs": Map {},
        },
        "fields": Object {
          "password": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "required": true,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
          "username": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "required": true,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
        },
        "tests": Array [],
        "transforms": Array [
          [Function],
        ],
      }
    }
  >
    <Component />
  </Formik>
</CenteredModal>
`;
